Sources/NIO/EventLoopFuture.swift

```diff
@@ -383,8 +383,14 @@ extension EventLoopFuture {
             switch self.value! {
             case .success(let t):
                 let futureU = callback(t)
-                return futureU._addCallback {
-                    return next._setValue(value: futureU.value!)
+                let otherEventLoop = futureU.eventLoop
+                if otherEventLoop.inEventLoop {
+                    return futureU._addCallback {
+                        return next._setValue(value: futureU.value!)
+                    }
+                } else {
+                    futureU.whenSuccess(next.succeed)
+                    return CallbackList()
```

Comment:
This will lose track of failures, which isn't what we want I don't think.

Reply:
doh, thanks!

Reply:
@weissi please add a test that will catch this ;)